src/Symfony/Component/Runtime/Runner/Middleware/MiddlewareFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Runtime\Runner\Middleware;
+
+/**
+ * @author Sascha Heilmeier <sascha.heilmeier@netlogix.de>
+ */
+readonly class MiddlewareFactory
+{
+    /**
+     * @param class-string<MiddlewareInterface> ...$middleware
+     * @return \Generator<MiddlewareInterface>
+     */
+    public function create(string ...$middleware): \Generator
```

Comment:
this should take an array instead of a variadic argument:

- the caller has an array anyway, so using a variadic argument is pure overhead by having to spread the array to rebuild another array
- a variadic argument means we don't have any way to make the signature evolve in the future (we cannot add a new argument at the end of the signature.

Reply:
I've implemented your suggestion